integration-tests/smb/src/test/java/org/apache/camel/quarkus/component/smb/it/SmbTestResource.java

```diff
@@ -0,0 +1,61 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.apache.camel.quarkus.component.smb.it;
+
+import java.util.Map;
+
+import io.quarkus.test.common.QuarkusTestResourceLifecycleManager;
+import org.testcontainers.containers.GenericContainer;
+import org.testcontainers.containers.wait.strategy.Wait;
+import org.testcontainers.images.builder.ImageFromDockerfile;
+
+public class SmbTestResource implements QuarkusTestResourceLifecycleManager {
+
+    private static final int SMB_PORT = 445;
+
+    private GenericContainer<?> container;
+
+    @Override
+    public Map<String, String> start() {
+        try {
+            container = new GenericContainer<>(new ImageFromDockerfile("localhost/samba:camel", false)
```

Comment:
So we use a locally built samba container ? Isn't there a kind of official image out there so that we don't need to maintain it ?

@jamesnetherton Probably we would like to avoid ImageFromDockerFile if possible ?

Reply:
> So we use a locally built samba container ? Isn't there a kind of official image out there so that we don't need to maintain it ?
> 
> @jamesnetherton Probably we would like to avoid ImageFromDockerFile if possible ?

It's a direct copy from how the Camel SMB test does things. It at least enables the test data to be set up easily.

There probably is an alternative. I'd have to check for a good candidate. Not sure if we need to do that now or follow up aftwards?

Reply:
Indeed, it maps the camel way. Guess it was done that way because there is no official image.
On the top of my head, we had also used ImageFromDockerFile in mail test where it worked for a while.
So probably, this is suitable for platform tests and also for people testing with different processors instruction sets.